### PR TITLE
ENH: Add annotations for `np.lib.npyio`

### PR DESCRIPTION
**Pull request by BvB93**
This PR adds annotations for the various functions in `np.lib.npyio`.

### CODE REVIEW

File: runtests.py [hunk 1]
```diff
@@ -476,13 +476,17 @@ def build_project(args):
...
+    ))
     site_dir = site_dir_template.format(platbase=dst_dir,
                                         py_version_short=py_v_s,
                                         platlibdir=platlibdir,
                                         base=dst_dir,
                                         )
-    noarch_template = sysconfig.get_path('purelib', expand=False)
+    noarch_template = os.path.normpath(sysconfig.get_path(
+        'purelib', expand=False
+    ))
```
Comment (BvB93) [inline runtests.py:489]: `sysconfig.get_path` always returns unix-style file-separators, which will result in mixed usage of `/` and `\` in windows paths (thus breaking one of the tests).
This change commit should definitely be backported.

File: numpy/lib/npyio.pyi [hunk 2]
```diff
@@ -1,98 +1,249 @@
...
+    newline: str = ...,
+    header: str = ...,
+    footer: str = ...,
+    comments: str = ...,
+    encoding: None | str = ...,
+) -> None: ...
+
+@overload
+def fromregex(
+    file: str | IO[Any],
```
Comment (BvB93) [inline numpy/lib/npyio.pyi:185]: `fromregex` is currently the only file-manipulation-function that does not support path-like objects.
Might be something to add in a future PR?

Comment (charris) [reply numpy/lib/npyio.pyi:185]: Maybe open an issue so it doesn't get lost.

**Comment by charris** [comment]
Thanks Bas.

File: numpy/lib/npyio.pyi (hunk 2 above)
Comment (BvB93) [reply numpy/lib/npyio.pyi:185]: Good call, I just created one at https://github.com/numpy/numpy/issues/19679.

**Pull request merged**